### ISSUE TITLE
Feature Request : Customisable rate limiting

### ISSUE THREAD

**Issue by sachinkamath** [CTFd/CTFd#1091]
### Background

I had to host a CTF in a college which had all the PC behind a NAT and around 300-350 were taking part in the CTF. Due to this, all the participants were getting rate-limited.

I tried to adjust the kpm present in [/api/v1/challenges#L380](https://github.com/CTFd/CTFd/blob/97f52756bc579d5aa8bdf5376cfbeb0700d5f91d/CTFd/api/v1/challenges.py#L380) and restarted the platform but that didn't seem to do the trick.

### Solution

It'd be nice to have a setting in the admin to customize the lockout or disable it altogether (maybe by setting it to zero). 

@ColdHeat if you can point me to the right lines where the lockouts for challenge attempts are implemented, I can work on this.

**Comment by ColdHeat** [comment]
This is implemented in the ratelimit decorator. https://github.com/CTFd/CTFd/blob/master/CTFd/utils/decorators/__init__.py#L130

**Comment by pandatix** [comment]
Up on this issue. Same problem here... 😅 
When dealing with configuration I have a ton of jobs that sets up in parallel, coming from the same host... That hits the `@ratelimit` decorator thus break mid-job.

One solution would have been to use an API key but in this very specific context it is not possible (it would in turn create a ton of API keys as they try to be idempotent scripts, from a fresh or already-configured CTFd instance, hence making any key management strategy impossible or lacky). Every alternatives I though about were just workarounds to a proper solution.

Do you want me to set up a PR to solve this @ColdHeat ?
Would you prefer it live-configurable (in the Config/Settings forms) or varenv for immutability ? Both are fine to me 🤷

**Comment by ColdHeat** [comment]
API rate limits should probably be defined in config and KPM can be defined in config with an override in the Admin Panel. 

A PR is okay.
